tempesta_fw/t/functional/tcp_connection/test_connection_close.py

```diff
@@ -0,0 +1,78 @@
+"""
+Tests for TCP connection closing.
+"""
+
+from __future__ import print_function
+from testers import functional
+from helpers import analyzer, deproxy, chains
+import asyncore
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2017 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+
+class CloseConnection(functional.FunctionalTest):
+    """Regular connection closing."""
+
+    def stop_and_close(self):
```

Comment:
It's just the same code as `functional.FunctionalTest.tearDown()` which is called after every test. Isn't it better to override `tearDown()` method and append `sniffer.stop()` there if it necessary?

Reply:
We need to close the connection to check (via `tcpdump`) the correctness of its closing (and in case of failure - to throw an exception in the `test*` function of class); so before stopping `tcpdump` and analyzing its output, we need to close `Deproxy` (and its client and server connections also)  - in `test*` function.

Reply:
Ok, I see the point. Then the function should be documented since the approach differs from other tests

Reply:
Added comment into `stop_and_close(self)` function.